code/modules/events/merchant/merchant_event.dm

```diff
@@ -0,0 +1,120 @@
+/datum/round_event_control/merchant
+	name = "Merchant"
+	typepath = /datum/round_event/merchant
+	max_occurrences = 1
+
+/datum/round_event_control/merchant/canSpawnEvent(players)
+	if(!SSeconomy.get_dep_account(ACCOUNT_CAR))
+		return FALSE //They can't pay?
+	if(!EMERGENCY_IDLE_OR_RECALLED)
+		return FALSE //merchant docks where escape is
+	if(istype(SSshuttle.emergency, /obj/docking_port/mobile/emergency/shuttle_build))
+		return FALSE //don't undo manual player engineering, it also would unload people and ghost them, there's just a lot of problems
+	return ..()
+
+/datum/round_event/merchant
+	var/datum/comm_message/merchant_message
+	var/datum/merchant/visiting_merchant
+
+/datum/round_event/merchant/announce(fake)
+	priority_announce("Incoming subspace communication. Secure channel opened at all communication consoles.", "Incoming Message", SSstation.announcer.get_rand_report_sound())
+
+/datum/round_event/merchant/setup()
+	var/merchant_path = pick(subtypesof(/datum/merchant))
+	visiting_merchant = new merchant_path
+
+/datum/round_event/merchant/start()
+	merchant_message = new(visiting_merchant, visiting_merchant.message_greet, list("Send [INITIAL_VISIT_COST] Credits.","Reject Offer."))
+	merchant_message.answer_callback = CALLBACK(src,.proc/answered)
+	SScommunications.send_message(merchant_message, unique = TRUE)
+
+/datum/round_event/merchant/proc/answered()
+	if(EMERGENCY_PAST_POINT_OF_NO_RETURN)
+		priority_announce(visiting_merchant.message_too_late, sender_override = visiting_merchant)
+		return
+	if(merchant_message?.answered == RESPONSE_MERCHANT_LEAVE)
+		return
+
+	var/datum/bank_account/station_balance = SSeconomy.get_dep_account(ACCOUNT_CAR)
+	if(!station_balance?.has_money(INITIAL_VISIT_COST))
+		priority_announce(visiting_merchant.message_too_poor, sender_override = visiting_merchant)
+		return
+	station_balance?.adjust_money(-INITIAL_VISIT_COST)
```

Comment:
`adjust_money` returns FALSE if you don't have enough money, you can combine these two.